src/hable.js

```diff
@@ -38,6 +38,20 @@ export default class Hookable {
 
     this._hooks[name] = this._hooks[name] || []
     this._hooks[name].push(fn)
+    
+    return () => {
+      if (fn) {
+        this.removeHook(name, fn)
+        fn = null // Free memory
+      }
+    }
+  }
+  
+  removeHook (name, fn) {
```

Comment:
Is it necessary to keep both callback fn and `removeHook`?

Reply:
Not necessary. I prefer the callback way, but depending on the use case a `removeHook` might be more elegant.

```js
someObj.hook('a', doStuff)
someObj.hook('b', doStuff)

const unreg = someObj.hable('c', () => {
  someObj.removeHook('a', doStuff)
  someObj.removeHook('b', doStuff)
  
  unreg()
})
```

What do you think ?

Reply:
It thinks it makes sense for different use cases. Only concern is duplicate logic introduced. As i plan to add browser target support that can be problematic.

Reply:
Also, because we do `fn = null` in the `addHook` callback, this allows to ensure that the `unreg` function actually only removes the callback once (only the one that was added initially):

```js
const unreg1 = someObj.hook('a', doStuff)
someObj.hook('a', doStuff)

unreg1()
unreg1()

// doStuff still once in the list of hooks
```

this is not the case with `removeHook`

```js
someObj.hook('a', doStuff)
someObj.hook('a', doStuff)

someObj.removeHook('a', doStuff)
someObj.removeHook('a', doStuff)

// doStuff no longer in the list of hooks
```

Both cases have their use.

Reply:
Any minifier would convert the return call to:
```js
return function(){(f&&this.removeHook(n,f));f=null}
```
51 bytes (uncompressed) is rly not much...

Reply:
The reason I like this pattern is that it allows writing more elegant code:
```js
const myCallback = () => {
  // doStuff
  hook.removeHook("test:hook", myCallback)
}
hook.hook("test:hook", myCallback)

// 67 bytes minified
```

vs

```js
const unreg = hook.hook("test:hook", () => {
  // doStuff
  unreg()
})

// 41 bytes minified
```

So by "wasting" some bytes in you lib, you actually help save bytes un user code